addons/web_editor/static/src/components/media_dialog/search_media.js

```diff
@@ -3,19 +3,28 @@
 import { debounce } from '@web/core/utils/timing';
 import { useAutofocus } from '@web/core/utils/hooks';
 
-import { Component, onWillUnmount, xml } from "@odoo/owl";
+import { Component, onWillUnmount, xml, useEffect, useState } from "@odoo/owl";
 
 export class SearchMedia extends Component {
     setup() {
         useAutofocus();
-        this.search = debounce((ev) => this.props.search(ev.target.value), 1000);
+        this.debouncedSearch = debounce(this.props.search, 1000);
+
+        this.state = useState({
+            input: '',
+        });
+
         onWillUnmount(() => {
             this.search.cancel();
         });
+
+        useEffect((input) => {
+            this.debouncedSearch(input);
+        }, () => [this.state.input]);
     }
 }
 SearchMedia.template = xml`
 <div class="position-relative mw-lg-25 flex-grow-1 me-auto">
-    <input type="text" class="o_we_search o_input form-control" t-att-placeholder="props.searchPlaceholder.trim()" t-att-value="props.needle" t-on-input="search" t-ref="autofocus"/>
+    <input type="text" class="o_we_search o_input form-control" t-att-placeholder="props.searchPlaceholder.trim()" t-model="state.input" t-ref="autofocus"/>
```

Comment:
Can you explain to me what this t-model does? :)

Reply:
https://github.com/odoo/owl/blob/master/doc/reference/input_bindings.md#-form-input-bindings-

Reply:
Ok, note to myself instead of having to re-read the full doc: `t-model` replace `t-on-input` by writing on the `state` automatically and the Component reacts to the state change via `useEffect`. And I guess that the writing of the value inside of the input is removed and we simply rely on the fact that what the user entered is what the state should be -> @younn-o you confirm? Writing on `state.input` does not update the input value?

Reply:
Yes, t-model is there to remove boilerplate _onInput handlers.
```js
class Input extends Component {
  setup() {
    this.state = useState({ input: '' });
  }

  _onInput(ev) {
    this.state.input = ev.target.value;
  }
}
Input.xml = `<input t-on-input="this._onInput"/>`;
```
becomes
```js
class Input extends Component {
  setup() {
    this.state = useState({ input: '' });
  }
}
Input.xml = `<input t-model="this.state.input"/>`;
```
When that state value is updated, we trigger the `search` callback prop (debounced) via an effect, so that the parent can process the value the user has entered.

I checked as it was not mentioned in the doc, updating the `state.input` value (used in `t-model`) will update the input value